src/github.com/log-yarder/yarder/storage/storage.go

```diff
@@ -0,0 +1,111 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+)
+
+const (
+	dirPerms  = 0777
+	filePerms = 777
+)
+
+// LogChunk represents a sequence of log entries.
+type LogChunk interface {
+	// Append add a single entry to an open chunk.
+	Append(entry string) error
+
+	// Size returns the current number of entries in the chunk.
+	Size() int
+
+	// Close declares an open chunk as done and performs necessary cleanup tasks.
+	Close() error
+}
+
+// Storage represents a single machine's interface to a corpus of almanac data.
+type Storage interface {
+	// CreateChunk allocates a new empty chunk of logs.
+	CreateChunk() (LogChunk, error)
+}
+
+// NewDiskStorage returns an instance of Strage backed by files on disk.
+func NewDiskStorage(rootPath string) Storage {
```

Comment:
This constructor (and the appender one too) are just stuffing the values into matching fields and initialising the rest to zero. Seems like the struct could be directly constructable then.

Reply:
Done.